api/api/controllers/search_controller.py

```diff
@@ -339,6 +347,18 @@ def build_search_query(
     )
 
 
+def log_query_features(query: str, query_name) -> None:
+    flags = [
+        ("PRECEDENCE", r"\(.*\)"),
+        ("ESCAPE", r"\\"),
+        ("FUZZY|SLOP", r"~\d"),
+        ("PREFIX", r"\*"),
+    ]
+    for flag, pattern in flags:
+        if bool(re.search(pattern, query)):
+            log.info(f"Special feature in `{query_name}` query string. {flag}: {query}")
```

Comment:
Just a suggestion, but it would produce less logs overall while making it easier to tally them if the logging did something like, counting the instances of each flag and logging all flags at once in JSON like `{ "log_message": "Special features present in query", "query_name": query_name, "query": query} | {flag: count_flag(flag, query) for flag in flags }` (one-liner just for commenting, not to suggest that's the way to do it).

No big deal, though. This way works too, it just produces slightly more logs (not many, because we know these are used so rarely and mostly following the frontend examples) and requires parsing the message string rather than using Logs Insights discovered fields feature.

Reply:
Thank you for this suggestion, I was actually not certain that the logging pattern would be good, and I didn't even realize that there could be several flags used in a single search. I'll update the logging

Reply:
Updated the logging, now if you go to `http://0.0.0.0:50280/v1/images/?q=cat+net*+(dog)+%5C(+someth~2`, you get the following message logged:
`[2023-11-30 08:32:53,795 - root - 362][INFO] [3419191a256b4058b2e8ffd5066b1ec2] {'log_message': 'Special features present in query', 'query_name': 'q', 'query': 'cat net* (dog) \\( someth~2', 'flags': ['PRECEDENCE', 'ESCAPE', 'FUZZY|SLOP', 'PREFIX']}`

I don't think we need to log the flag cound separately, do you, @sarayourfriend ?

Reply:
I guess the raw count isn't that important, is it! Good call :+1: The change LGTM.